dali/operators/sequence/sequence_rearrange.cu

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2019-2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <utility>
+#include <vector>
+
+#include "dali/kernels/common/scatter_gather.h"
+#include "dali/operators/sequence/sequence_rearrange.h"
+
+namespace dali {
+
+template <>
+void SequenceRearrange<GPUBackend>::RunImpl(workspace_t<GPUBackend> &ws) {
+  scatter_gather_.Reset();
+  const auto &input = ws.InputRef<GPUBackend>(0);
+  auto &output = ws.OutputRef<GPUBackend>(0);
+  TypeInfo type = input.type();
+
+  for (int sample_idx = 0; sample_idx < batch_size_; ++sample_idx) {
+    auto *out_sample = reinterpret_cast<char *>(output.raw_mutable_tensor(sample_idx));
+    const auto *in_sample = reinterpret_cast<const char *>(input.raw_tensor(sample_idx));
+    const auto &in_shape = input.shape()[sample_idx];
+    auto element_sizeof = volume(in_shape.last(in_shape.sample_dim() - 1)) * type.size();
+
+    TensorView<StorageCPU, const int, 1> new_order = {};
+
+    if (single_order_) {
+      new_order = TensorView<StorageCPU, const int, 1>(new_order_.data(),
+                                                       TensorShape<1>(new_order_.size()));
+    } else {
+      auto &new_orders = ws.ArgumentInput("new_order");
```

Comment:
I don't see any validation of `new_order` values.

Reply:
it's in `GetOutputShape`, I test if it's between [0, input_length), in SetupImpl I check if it's not empty.

Reply:
Missed that.

Reply:
I'll not block this PR over this, but I think that the fact we both missed that signalizes a design problem - at least it was so with me: GetOutputShape doesn't need to look at the indices to do its job, so I didn't look for index verification there - indeed, I'd be surprised to find out that this function needs actual indices, not just their number.
If it's not a big problem, I'd appreciate if the two functionalities were separated.